### PR TITLE
feat(l2): write contracts' addresses to .env file

### LINKED ISSUE #1072: L2: Update `.env` after contracts are deployed in the base layer

### PR DESCRIPTION
**Pull request by fborello-lambda**
**Motivation**

After the contracts are deployed, the addresses will differ from the `.env.example` provided. To make the deployment process more automatic, the `.env` file addresses should be changed.

**Description**

- Read `.env` file in order to change `PROPOSER_ON_CHAIN_PROPOSER_ADDRESS` and `L1_WATCHER_BRIDGE_ADDRESS ` with its corresponding addresses.

Closes #1072

### CODE REVIEW

**Review by jrchatruc — APPROVED** [review]

File: crates/l2/contracts/deployer.rs [hunk 1]
```diff
@@ -31,6 +31,27 @@ async fn main() {
...
+    let env_lines = read_env_as_lines().expect("Failed to read env file as lines.");
+
+    let mut wr_lines: Vec<String> = Vec::new();
+    for line in env_lines {
+        let mut line = line.unwrap();
+        if let Some(eq) = line.find('=') {
+            let (envar, _) = line.split_at(eq);
+            line = match envar {
+                "PROPOSER_ON_CHAIN_PROPOSER_ADDRESS" => {
+                    format!("{}={:?}", envar, on_chain_proposer)
```
Comment (ilitteri) [inline crates/l2/contracts/deployer.rs:44]:
```suggestion
                    format!("{envar}={on_chain_proposer:#x}")
```

File: crates/l2/contracts/deployer.rs [hunk 2]
```diff
@@ -31,6 +31,27 @@ async fn main() {
...
+                }
+                "L1_WATCHER_BRIDGE_ADDRESS" => {
+                    format!("{}={:?}", envar, bridge_address)
```
Comment (ilitteri) [inline crates/l2/contracts/deployer.rs:47]:
```suggestion
                    format!("{envar}={bridge_address:#x}")
```

File: crates/l2/utils/config/mod.rs [hunk 3]
```diff
@@ -34,3 +34,28 @@ pub fn read_env_file() -> Result<(), errors::ConfigError> {
 
     Ok(())
 }
+
+pub fn read_env_as_lines(
+) -> Result<std::io::Lines<std::io::BufReader<std::fs::File>>, errors::ConfigError> {
+    let env_file_name = std::env::var("ENV_FILE").unwrap_or_else(|_| ".env".to_string());
```
Comment (ilitteri) [inline crates/l2/utils/config/mod.rs:40]: Can't we use `unwrap_or` here?
```suggestion
    let env_file_name = std::env::var("ENV_FILE").unwrap_or(".env".to_owned());
```

Comment (ilitteri) [reply crates/l2/utils/config/mod.rs:40]: Maybe we can have a `DEFAULT_ENV_FILE_NAME` constant.

File: crates/l2/utils/config/mod.rs [hunk 4]
```diff
@@ -34,3 +34,28 @@ pub fn read_env_file() -> Result<(), errors::ConfigError> {
...
+    let env_file = std::fs::File::open(env_file_name)?;
+    let reader = std::io::BufReader::new(env_file);
+
+    Ok(reader.lines())
+}
+
+pub fn write_env(lines: Vec<String>) -> Result<(), errors::ConfigError> {
+    let env_file_name = std::env::var("ENV_FILE").unwrap_or_else(|_| ".env".to_string());
```
Comment (ilitteri) [inline crates/l2/utils/config/mod.rs:48]: Ditto.

File: crates/l2/utils/config/mod.rs [hunk 5]
```diff
@@ -34,3 +34,28 @@ pub fn read_env_file() -> Result<(), errors::ConfigError> {
...
+
+    let file = std::fs::OpenOptions::new()
+        .write(true)
+        .truncate(true)
+        .open(&env_file_name)?;
+
+    let mut writer = std::io::BufWriter::new(file);
+    for line in lines {
+        writeln!(writer, "{}", line)?;
```
Comment (ilitteri) [inline crates/l2/utils/config/mod.rs:57]:
```suggestion
        writeln!(writer, "{line}")?;
```

**Review by ilitteri — COMMENTED** [review]
LGTM. Left some suggestions

File: crates/l2/utils/config/mod.rs (hunk 4 above)
Comment (fborello-lambda) [reply crates/l2/utils/config/mod.rs:48]: Yes, the default name for the envfile is `.env` the `envar` can be used to set the non-relative path if necessary.

Comment (ilitteri) [reply crates/l2/utils/config/mod.rs:48]: I got this. By the previous comment I meant that we could have a `const DEFAULT_ENV_FILE_NAME: &str = ".env";` and then use `unwrap_or(DEFAULT_ENV_FILE_NAME.to_owned())`

**Pull request merged**